torch/_inductor/utils.py

```diff
@@ -45,6 +45,79 @@
 VarRanges = Dict[sympy.Expr, sympy.Expr]
 
 
+def do_bench_using_profiling(fn: Callable[[], Any], warmup=25, rep=100) -> float:
```

Comment:
Why can't the existing do_bench be used?  This seems pretty similar.

Reply:
do_bench also measures CPU-side overhead.

e.g. for an event sequence:
CUDA event begin -> CUDA launch kernel -> CUDA kernel execution -> CUDA event end
It measures time between CUDA event begin and CUDA event end, which contains "CUDA launch kernel" part. This part could take some CPU time. This is especially bad for CUTLASS kernels which rely on ctypes to invoke C++ functions from Python.

do_bench_using_profiling, on the other hand, relies on profiler to collect kernel device time.